### PR TITLE
Infer static code analysis

### PR DESCRIPTION
**Pull request by rigomate**
Add infer static code analysis checker to builds.
Solve one trivial possible null pointer dereferencing.

### CODE REVIEW

**Comment by rigomate** [comment]
The infer action gives such an overview of the issues, which it suspects:
image

**Comment by Brutus5000** [comment]
If we run it in the CI where do we see the results? Your screenshot indicates it's somewhere on Github?

**Comment by Sheikah45** [comment]
Also how is this different from codacy which we use in other repositories

**Comment by rigomate** [comment]
It can be seen in the actions section at github:
In my fork it is here: https://github.com/rigomate/java-ice-adapter/actions/runs/6855882675

I don't know how it differs codacy, but my philosophy is simple: use as many code checkers as possible. Maybe the will find some viable issue.
I think this null pointer call which it found and is fixed in the pull request sets a good example, that this tool might find real issues.

Cheers!

File: ice-adapter/src/main/java/com/faforever/iceadapter/gpgnet/GPGNetServer.java [hunk 1]
```diff
@@ -238,8 +238,8 @@ public static Optional<GameState> getGameState() {
      */
     public static void close() {
         if (currentClient != null) {
-            currentClient = null;
             currentClient.close();
+            currentClient = null;
```
Comment (rigomate) [inline ice-adapter/src/main/java/com/faforever/iceadapter/gpgnet/GPGNetServer.java:242]: Even if infer checking is not wished, this should be fixed, right?

**Review by Brutus5000 — APPROVED** [review]
Why not give it a try :)

**Pull request merged**